File: src/main/java/me/beanbag/nuker/module/modules/canaltools/IProtectionLocator.java
package me.beanbag.nuker.module.modules.canaltools;

import net.minecraft.block.BlockState;

import java.util.List;

public interface IProtectionLocator {
    boolean isProtection(int x, int y, int z);
    boolean isCorrectInPosition(int x, int y, int z);
    List<BlockState> statesForPosition(int x, int y, int z);
}

File: src/main/kotlin/mc/merge/module/modules/canal/tools/IProtectionLocator.kt
package mc.merge.module.modules.canal.tools

import net.minecraft.block.BlockState

interface IProtectionLocator {
    fun isProtection(x: Int, y: Int, z: Int): Boolean
    fun isCorrectInPosition(x: Int, y: Int, z: Int): Boolean
    fun statesForPosition(x: Int, y: Int, z: Int): List<BlockState?>?
}
